typhon/signal.py

```diff
@@ -107,6 +108,11 @@ def signal_widget(signal, read_only=False, tooltip=None):
     widget_instance.setObjectName(name)
     if tooltip is not None:
         widget_instance.setToolTip(tooltip)
+    try:
```

Comment:
I think this line could be be moved up to line 93. We don't really want to set this on either of the plotting options.

Reply:
Are you saying that you want to prevent these string signals from being plotted?

Reply:
No, the `widget_instance` that is created can either be `PyDMLineEdit` / `PyDMLabel` or a `QPushButton` in the case of arrays. For the latter setting `widget_instance.display_format` does nothing. Where you have it in the current code sets it regardless of the type of `widget_instance` we create.

Sorry this function is a mess. It kind of grew organically and is due for a refactor.

Reply:
Got it, though I assume you mean `QComboBox`. I'm assuming this throws an exception then in the case of the combo box?

Reply:
I didn't, but that is also true. A `QPushButton` gets created for images and waveforms that allows you to look at the signal in a dialog.

Reply:
This whole block gets simpler if we switch to `PyDMLineEdit` in read-only mode rather than use the label... 🤔